lib/AST/Builtins.cpp

```diff
@@ -1198,8 +1198,15 @@ static ValueDecl *getAutoDiffApplyDerivativeFunction(
         SmallVector<FunctionType::Param, 2> params;
         for (auto &paramGen : fnParamGens)
           params.push_back(FunctionType::Param(paramGen.build(builder)));
+        // FIXME: This code used to implicitly default the following
+        // ExtInfo state, which may be correct or it may just happen
+        // to work. If you understand this code, then please audit
+        // it and please explain in a comment why either the default
+        // ExtInfo state is actually correct or why you changed the
+        // code to set/propagate some state. Thanks -- DAZ
+        FunctionType::ExtInfo info;
```

Comment:
Why the stack allocation rather than inlined into the call?  Not that it will make a difference in codegen, but may make the diff easier to read.

Reply:
For various reasons, I'm not a fan of statements/expressions that overflow a line so this is what felt natural to me. If you feel strongly about this, then I can change the pattern used.